packages/manager/test/__testutils__/mockPrismicRepositoryAPI.ts

```diff
@@ -0,0 +1,203 @@
+import { TestContext } from "vitest";
+import { rest } from "msw";
+
+type MockPrismicAuthAPIConfig = {
+	endpoint?: string;
+	existsEndpoint?: {
+		isSuccessful?: boolean;
+		expectedAuthenticationToken: string;
+		expectedCookies: string[];
+		domain: string;
+		exists: boolean;
+	};
+	newEndpoint?: {
+		isSuccessful?: boolean;
+		expectedAuthenticationToken: string;
+		expectedCookies: string[];
+		domain: string;
+		framework?: string;
+	};
+	deleteEndpoint?: {
+		isSuccessful?: boolean;
+		expectedAuthenticationToken: string;
+		expectedCookies: string[];
+		domain: string;
+		password: string;
+	};
+	starterDocumentsEndpoint?: {
+		isSuccessful?: boolean;
+		failureReason?: string;
+		expectedAuthenticationToken: string;
+		expectedCookies: string[];
+		domain: string;
+		signature: string;
+		documents: Record<string, unknown>;
+	};
+};
+
+export const mockPrismicRepositoryAPI = (
+	ctx: TestContext,
+	config: MockPrismicAuthAPIConfig,
+): void => {
+	const endpoint = config.endpoint ?? "https://prismic.io/";
+
+	if (config.existsEndpoint) {
+		ctx.msw.use(
+			rest.get(
+				new URL(
+					`./app/dashboard/repositories/${config.existsEndpoint.domain}/exists`,
+					endpoint,
+				).toString(),
+				(req, res, ctx) => {
+					if (
+						(config.existsEndpoint?.isSuccessful ?? true) &&
+						req.headers.get("Authorization") ===
+							`Bearer ${config.existsEndpoint?.expectedAuthenticationToken}` &&
+						req.headers.get("Cookie") ===
+							config.existsEndpoint?.expectedCookies.join("; ") &&
+						req.headers.get("User-Agent") === "slice-machine"
+					) {
+						// The real API returns "false" if the
+						// repository exists (i.e. it is the
+						// opposite of what you might expect).
+
+						if (config.existsEndpoint?.exists) {
+							return res(ctx.text("false"));
+						} else {
+							return res(ctx.text("true"));
+						}
+					} else {
+						return res(ctx.status(500));
+					}
+				},
+			),
+		);
+	}
+
+	if (config.newEndpoint) {
+		ctx.msw.use(
+			rest.post(
+				new URL("./authentication/newrepository", endpoint).toString(),
+				async (req, res, ctx) => {
+					if (req.url.searchParams.get("app") !== "slicemachine") {
+						return res(ctx.text(""), ctx.status(500));
+					}
+
+					if (config.newEndpoint?.isSuccessful ?? true) {
+						const body = await req.json();
+
+						const isExpectedBody =
+							body.plan === "personal" &&
+							body.isAnnual === "false" &&
+							body.role === "developer" &&
+							body.domain === config.newEndpoint?.domain &&
+							(config.newEndpoint?.framework
+								? body.framework === config.newEndpoint?.framework
+								: true);
+
+						if (
```

Comment:
Like an early exit pattern

Reply:
I prefer the `if...else` statements over guards because it shows the different branches more explicitly. Spreading multiple `returns` throughout the function can be difficult to trace.

However, I agree; there is a lot of nesting happening here.

This is what a version of this request handler looks like using guards:

```ts
rest.post(
	new URL("./authentication/newrepository", endpoint).toString(),
	async (req, res, ctx) => {
		if (req.url.searchParams.get("app") !== "slicemachine") {
			return res(ctx.text(""), ctx.status(500));
		}

		if (config.newEndpoint?.isSuccessful === false) {
			return res(ctx.text(""), ctx.status(500));
		}

		const body = await req.json();

		const isExpectedBody =
			body.plan === "personal" &&
			body.isAnnual === "false" &&
			body.role === "developer" &&
			body.domain === config.newEndpoint?.domain &&
			(config.newEndpoint?.framework
				? body.framework === config.newEndpoint?.framework
				: true);

		if (
			!isExpectedBody ||
			req.headers.get("Authorization") !==
				`Bearer ${config.newEndpoint?.expectedAuthenticationToken}` ||
			req.headers.get("Cookie") !==
				config.newEndpoint?.expectedCookies.join("; ") ||
			req.headers.get("User-Agent") !== "prismic-cli/sm"
		) {
			return res(ctx.text(""), ctx.status(500));
		}

		return res(
			ctx.text(config.newEndpoint?.domain || ""),
			ctx.status(200),
		);
	},
),
```

WDYT? Better? The same?